test/test_main.py

```diff
@@ -792,8 +809,10 @@ def _call_event(self, rel_path, env):
 
     def test_setup_event_links(self):
         """Test auto-creation of symlinks caused by initial events."""
-        all_event_hooks = [f"hooks/{e.replace('_', '-')}"
-                           for e in self.charm_module.Charm.on.events().keys()]
+        all_event_hooks = [f"hooks/{name.replace('_', '-')}"
+                           for name, event_source in self.charm_module.Charm.on.events().items()
+                           if not event_source.event_type.__name__ == "CustomEvent"]
```

Comment:
Nit: `!= "CustomEvent"` would be a bit simpler.

However, could we do something like this instead, to avoid matching on the type name?

    if issubclass(event_source.event_type, FrameworkEvent)